fri/src/two_adic_pcs.rs

```diff
@@ -389,18 +389,29 @@ where
                     .map(|&height| Dimensions { width: 0, height })
                     .collect_vec();
 
-                let batch_max_height = batch_heights.iter().max().expect("Empty batch?");
-                let log_batch_max_height = log2_strict_usize(*batch_max_height);
-                let bits_reduced = log_global_max_height - log_batch_max_height;
-                let reduced_index = index >> bits_reduced;
-
-                self.mmcs.verify_batch(
-                    batch_commit,
-                    &batch_dims,
-                    reduced_index,
-                    &batch_opening.opened_values,
-                    &batch_opening.opening_proof,
-                )?;
+                if let Some(batch_max_height) = batch_heights.iter().max() {
+                    let log_batch_max_height = log2_strict_usize(*batch_max_height);
+                    let bits_reduced = log_global_max_height - log_batch_max_height;
+                    let reduced_index = index >> bits_reduced;
+
+                    self.mmcs.verify_batch(
+                        batch_commit,
+                        &batch_dims,
+                        reduced_index,
+                        &batch_opening.opened_values,
+                        &batch_opening.opening_proof,
+                    )?;
+                } else {
+                    // Empty batch?
+                    self.mmcs.verify_batch(
```

Comment:
this just passes error handling to `verify_batch`, which is going to return an error anyways right? isn't it clearer to just return an error directly

Reply:
Yeah... in principle some or all `Mmcs`s could support empty batches, but the current `MerkleTreeMmcs` doesn't. We could add support, but it'd be a bit of extra code and I'm not sure if we'll ever have a case where it's useful.

It seems like this change is leaving the door open to the possibility that some or all `Mmcs`s could support empty batches later, but we could also declare empty batches are invalid for now; either option seems fine to me.

Reply:
I see - I forgot here the `Mmcs` is a trait (even though there's only one implementation). Seems ok to leave it then.

Reply:
Yep you got it.